public/app/core/components/TimelineChart/utils.ts

```diff
@@ -162,29 +162,33 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<UPlotConfigOptions> = (
     range: coreConfig.yRange,
   });
 
+  const xAxisHidden = frame.fields[0].config.custom.axisPlacement === AxisPlacement.Hidden;
+
   builder.addAxis({
+    show: !xAxisHidden,
     scaleKey: xScaleKey,
     isTime: true,
     splits: coreConfig.xSplits!,
     placement: AxisPlacement.Bottom,
     timeZone: timeZones[0],
     theme,
-    grid: { show: true },
   });
 
-  const yAxisWidth = frame.fields[1].config.custom.axisWidth;
+  const yCustomConfig = frame.fields[1].config.custom;
+  const yAxisWidth = yCustomConfig.axisWidth;
+  const yAxisHidden = yCustomConfig.axisPlacement === AxisPlacement.Hidden;
```

Comment:
would we want `yCustomConfig.axisPlacement === AxisPlacement.Hidden || yAxisWidth === 0` ? not super likely but we'd have consistency

Reply:
the idea is that we don't hide the labels in the width=0 case, we just dont have a way to align them currently to the left edge of the viz and they end up being rendered off screen. once we fix the alignment issue, we can use `width === 0` to change the label alignment and they will render over the viz like old Discrete panel.

Reply:
sounds good, nevermind then!

Reply:
![image](https://github.com/user-attachments/assets/eebcec3b-2ede-435e-8c70-7ef8c737e40e)